### PR TITLE
Extend Network doc comment for other networks

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change
Extend Network doc comment for other networks, because without it it's not clear why it might not work

## Type of change

- [x] Documentation Fix

### CODE REVIEW

**Comment by Thoralf-M** [comment]
If you don't use .with_network() for the ClientBuilder then it's not needed to match and the only difference would be that also nodes from another network, for example if you provide "mainnet" and a single testnet node, then it would just use this testnet node and now it would return an error that no node is available or ` Failed to get an answer from all nodes`

**Comment by cycraig** [comment]
I would argue that we still want that behaviour with `.with_network()` to prevent mismatches in where DIDs are published/resolved, e.g. we don't want a `"did:iota:test:xyz"` DID published on the mainnet, nor ones with a custom network name? I acknowledge that it might be confusing for new developers when it doesn't work with only a cryptic error message, as per Discord, so the restrictions on DID network names should be relaxed a bit to match those of private Tangle nodes (at least "private-tangle" for starters).

File: identity-iota/src/tangle/network.rs [hunk 1]
```diff
@@ -39,7 +39,8 @@ impl Network {
   ///
   /// The names `"test"` and `"main"` will be mapped to the well-known [`Testnet`][Network::Testnet]
   /// and [`Mainnet`][Network::Mainnet] networks respectively. Other inputs will return an instance
-  /// of [`Other`][Network::Other] if the name is valid.
+  /// of [`Other`][Network::Other] if the name is valid. It must match a part or all of the networkId returned in the
+  /// nodeinfo from a node. For exampe, if the networkId is `"private-tangle"`, `"tangle"` can be used.
```
Comment (cycraig) [inline identity-iota/src/tangle/network.rs:43]:
```suggestion
  /// of [`Other`][Network::Other] if the name is valid. It must match a part or all of the `networkId` returned in the
  /// nodeinfo from a node. For example, if the networkId is `"private-tangle"`, `"tangle"` can be used.
```

**Review by cycraig — APPROVED** [review]
Spelling fix otherwise looks good to me. Thanks for the clarification.

I'll open another PR for any changes to the network name pending discussions.

**Pull request merged**